packages/platform/src/response.rs

```diff
@@ -1,12 +1,31 @@
+use serde::Serialize;
+
 use sdk::{
-    cosmwasm_ext::Response,
-    cosmwasm_std::{to_binary, StdResult},
+    cosmwasm_ext::Response as CwResponse,
+    cosmwasm_std::{to_binary, Binary, StdResult},
 };
-use serde::Serialize;
 
-pub fn response<T>(resp: &T) -> StdResult<Response>
+#[inline]
+pub fn response<T>(response: &T) -> StdResult<Response>
 where
     T: Serialize + ?Sized,
 {
-    Ok(Response::new().set_data(to_binary(resp)?))
+    response_with_messages(CwResponse::new(), response)
+}
+
+pub fn response_with_messages<T, U>(messages: T, response: &U) -> StdResult<Response>
```

Comment:
cosmetic: change the order of fn parameters to reflect the fn name, first the response, then the messages